### DISCUSSION TITLE
Unauthorized

### DISCUSSION THREAD

**Discussion by hdswiss** [wg-easy/wg-easy#1451] [Q&A] [answered]
"I have installed WireGuard Easy on Synology. When I go to http://xxxxxxxx.synology.me:51821/, I cannot log in. On http://xxxxxxxx.synology.me:51821, the following is displayed: Unauthorized. The password is correct."
<img width="338" alt="Screenshot 2024-10-06 002047" src="https://github.com/user-attachments/assets/3759c6e5-1dcf-4201-9bcd-21d21532351b">

**Comment by daniroxas** [comment]
most likely you copied the HASH PASSWORD incorrectly into the script

**Comment by hdswiss** [comment]
The password is 100% correct, but I still can't log in.

> docker run -d \
  --name wg-easy \
  --env LANG=de \
  --env WG_HOST=xxxxxxxx.synology.me \
  --env PASSWORD_HASH=1258trea65 \
  --env PORT=51821 \
  --env WG_PORT=51820 \
  -v /volume1/docker/wg-easy:/etc/wireguard \
  --publish 51820:51820/udp \
  --publish 51821:51821/tcp \
  --cap-add NET_ADMIN \
  --cap-add SYS_MODULE \
  --sysctl 'net.ipv4.conf.all.src_valid_mark=1' \
  --sysctl 'net.ipv4.ip_forward=1' \
  --restart unless-stopped \
  ghcr.io/wg-easy/wg-easy:latest

**Reply by daniroxas** [reply]
hash must be copied from ' ' example: '$atdufur466....45'

**Reply by daniroxas** [reply, marked as answer]
https://github.com/wg-easy/wg-easy/blob/master/How_to_generate_an_bcrypt_hash.md

**Reply by FelixSche** [reply]
> https://github.com/wg-easy/wg-easy/blob/master/How_to_generate_an_bcrypt_hash.md

Link is dead

**Reply by kaaax0815** [reply]
https://github.com/wg-easy/wg-easy/blob/production/How_to_generate_an_bcrypt_hash.md

**Comment by hdswiss** [comment]
Sorry, but where do I need to add the password hash?

> root@MYNAS:~# docker run -it ghcr.io/wg-easy/wg-easy:latest wgpw 1258trea65
PASSWORD_HASH='$2a$12$k5NM84DtSvCwEpIyP.FLxu5rXuD/HHKg9KjY4oQ62ZAyA3T7qqJgO'

**Reply by daniroxas** [reply]
docker run --detach \
  --name wg-easy \
  --env LANG=de \
  --env WG_HOST= xxxxxxxx.synology.me \
  --env PASSWORD_HASH='$2a$12$k5NM84DtSvCwEpIyP.FLxu5rXuD/HHKg9KjY4oQ62ZAyA3T7qqJgO' \
  --env PORT=51821 \
  --env WG_PORT=51820 \
  --volume ~/.wg-easy:/etc/wireguard \
  --publish 51820:51820/udp \
  --publish 51821:51821/tcp \
  --cap-add NET_ADMIN \
  --cap-add SYS_MODULE \
  --sysctl 'net.ipv4.conf.all.src_valid_mark=1' \
  --sysctl 'net.ipv4.ip_forward=1' \
  --restart unless-stopped \
  ghcr.io/wg-easy/wg-easy

**Comment by hdswiss** [comment]
Found it, now it works. Thank you
